src/module/actor/character/crafting/entry.ts

```diff
@@ -28,24 +32,39 @@ export class CraftingEntry implements CraftingEntryData {
         this.batchSize = data.batchSize;
         this.fieldDiscoveryBatchSize = data.fieldDiscoveryBatchSize;
         this.craftableItems = data.craftableItems;
-        this.preparedFormulas = knownFormulas.filter((formula) => formula.preparedData[this.selector] !== undefined);
-        this.tidyUpSlotData();
+        this.preparedFormulaData = (data.preparedFormulaData || [])
+            .map((prepData) => {
+                const formula = knownFormulas.find((formula) => formula.uuid === prepData.itemUUID);
+                if (formula) return prepData;
+                return null;
+            })
+            .filter((prepData): prepData is PreparedFormulaData => !!prepData);
+        this.parentItem = data.parentItem || {};
```

Comment:
The types doesn't blow up here only because `data.parentItem` is not-nullish. Why the `|| {}`?